gmso/external/convert_mol2.py

```diff
@@ -0,0 +1,175 @@
+"""Convert to and from a TRIPOS mol2 file."""
+# TODO add sources of mol2 files
+import os
+import warnings
+
+import unyt as u
+
+from gmso import Atom, Bond, Box, Topology
+from gmso.core.element import element_by_name, element_by_symbol
+
+
+def from_mol2(
+    filename, site_type="Atom"
+):  # TODO add flags for information to return
+    """Read in a TRIPOS mol2 file format into a gmso topology object.
+
+    Creates a Topology from a mol2 file structure. This will read in the
+    topological structure (sites, angles, and box) information into gmso.
+    Note that parameterized information can be found in these objects, but
+    will not be converted to the Topology.
+
+    Parameters
+    ----------
+    filename : string
+        path to the file where the mol2 file is stored.
+    site_type : string, optional, default='Atom'
+        tells the reader to consider the elements saved in the mol2 file, and
+        if the type is 'lj', to not try to identify the element of the site,
+        instead saving the site name.
+
+    Returns
+    -------
+    top : gmso.Topology
+
+    Notes
+    -----
+    It may be common to want to create an mBuild compound from a mol2 file. This is possible
+    by installing [mBuild](https://mbuild.mosdef.org/en/stable/index.html)
+    and converting using the following python code:
+
+        >>> from gmso.external.convert_mol2 import from_mol2
+        >>> from gmso.external.convert_mbuild import to_mbuild
+        >>> top = from_mol2('myfile.mol2')
+        >>> mbuild_compound = to_mbuild(top)
+    """
+    msg = "Provided path to file that does not exist"
+    if not os.path.isfile(filename):
+        raise OSError(msg)
+    # Initialize topology
+    topology = Topology(name=os.path.splitext(os.path.basename(filename))[0])
+    # save the name from the filename
+    f = open(filename, "r")
+    line = f.readline()
+    while f:
+        # check for header character in line
+        if line.startswith("@<TRIPOS>"):
+            # if header character in line, send to a function that will direct it properly
+            line, topology = parse_record_type_indicator(
```

Comment:
What is the motivation behind returning these lines? and the Topology?

Reply:
Totally unnecessary, will remove passing both of these around since they're modified in place.

Reply:
So I've been playing around with it, I think I need to keep passing the line around with the current usage. If I keep reading until the next rti, then I'll need to hold on to that line to see which function needs to draw that data from in the future. Topology passing is removed though.